src/com/carolinarollergirls/scoreboard/json/AutoSaveJSONState.java

```diff
@@ -0,0 +1,165 @@
+        File tmp = null;
+        FileWriter out = null;
+        try {
+            String json = JSON.std
+                          .with(JSON.Feature.PRETTY_PRINT_OUTPUT)
+                          .composeString()
+                          .startObject()
+                          .putObject("state", new TreeMap<>(jsm.getState()))
+                          .end()
+                          .finish();
+            tmp = File.createTempFile(file.getName(), ".tmp", dir);
+            out = new FileWriter(tmp);
+            out.write(json);
+            out.close();
+            tmp.renameTo(file); // This is atomic.
+        } catch (Exception e) {
+            ScoreBoardManager.printMessage("Error writing JSON autosave: " + e.getMessage());
+        } finally {
+            if (out != null) {
+                try { out.close(); } catch (Exception e) { }
+            }
+            if (tmp != null ) {
+                try { tmp.delete(); } catch (Exception e) { }
+            }
+        }
+    }
+
+    private void backupAutoSavedFiles() {
+        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd-HH-mm-ss");
+        File mainBackupDir = new File(dir, "backup");
+        File backupDir = new File(mainBackupDir, dateFormat.format(new Date()));
+        if (backupDir.exists()) {
+            ScoreBoardManager.printMessage("Could not back up auto-save files, backup directory already exists");
+        } else {
+            int n = 0;
+            do {
+                File from = getFile(n);
+                if (from.exists()) {
+                    try {
+                        FileUtils.copyFileToDirectory(from, backupDir, true);
+                    } catch ( Exception e ) {
+                        ScoreBoardManager.printMessage("Could not back up auto-save file '"+from.getName()+"' : "+e.getMessage());
+                    }
+                }
+            } while (n++ < AUTOSAVE_FILES);
+        }
+    }
+
+    public File getFile(int n) {
+        return getFile(n, dir);
+    }
+    public static File getFile(int n, File dir) {
+        return new File(dir, ("scoreboard-" + (n * INTERVAL_SECONDS) + "-secs-ago.json"));
+    }
+
+    public static boolean loadAutoSave(ScoreBoard scoreBoard, File dir) {
+        for (int i=0; i <= AUTOSAVE_FILES; i++) {
+            File f = getFile(i, dir);
+            if (!f.exists()) {
+                continue;
+            }
+            try {
+              loadFile(scoreBoard, f);
+                ScoreBoardManager.printMessage("Loaded auto-saved scoreboard from "+f.getPath());
+                return true;
+              } catch ( Exception e ) {
+                ScoreBoardManager.printMessage("Could not load auto-saved scoreboard XML file "+f.getPath()+" : "+e.getMessage());
+                e.printStackTrace();
+              }
+            }
+
+        return false;
+    }
+ 
+    public static void loadFile(ScoreBoard scoreBoard, File f) throws Exception{
```

Comment:
This mostly duplicates code from `LoadJsonScoreBoard` - maybe that should be factored into `ScoreBaordJSONSertter`

Reply:
Done